R/finalsize_risk_grps.R

```diff
@@ -76,6 +101,31 @@ final_size_grps <- function(contact_matrix,
       )
   )
 
+  # check which solver is requested
+  solver <- match.arg(arg = solver, several.ok = FALSE)
+
+  f_solver <- switch(solver,
+    newton = solve_final_size_newton,
+    iterative = solve_final_size_iterative
+  )
+
+  # prepare default list of control params
+  con <- list(
+    iterations = 1000,
+    tolerance = 1e-6,
+    step_rate = 1.9,
+    adapt_step = TRUE
+  )
+  # assign user specified values
+  con[(names(control))] <- control
+  extra_names <- names(control)[!names(control) %in% names(con)]
+  if (length(extra_names)) {
+    warning("Unknown names in control: ", paste(extra_names, collapse = ", "))
+  }
+
+  # prune control list if using newton solver
+  if (solver == "newton") con <- con[c("iterations", "tolerance")]
```

Comment:
I think it'll be easier to handle the default settings and argument checking directly in the solver functions.

Reply:
I just scrolled down a bit more and it looks like this is already done so I believe L112 to 127 can safely be removed.

Reply:
I'm not sure what you mean, but commenting out L112 - 127 and substituting `control` for `con` at L.147 does pass tests so I'll change that.